pkg/agent/clean/dupe_bindings.go

```diff
@@ -180,27 +180,19 @@ func (bc *dupeBindingsCleanup) cleanObjectDuplicates(bindingType string, newLabe
 			var CRBduplicates, RBDupes int
 
 			crbs, err := bc.clusterRoleBindings.List(metav1.ListOptions{LabelSelector: label})
-			if err != nil {
-				returnErr = multierror.Append(returnErr, err)
-			}
+			returnErr = errors.Join(returnErr, err)
```

Comment:
If you wanted to make it the same as before, you'd have to do it this way:

```go
var returnErr []error
// A couple of times
... returnErr = append(returnErr, err)
// Then when you want an error
return errors.Join(returnErr...)
```

That's because hashicorp `multierror.Append` would detect if it was a `multierror.Error` object being passed in and would append the error to it if it was. Otherwise, it would create a new `multierror.Error`. See https://github.com/hashicorp/go-multierror/blob/main/append.go#L14.

However the Go version doesn't do that. See https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/errors/join.go;l=19. Basically, the Go version does create a `joinError` object but it doesn't detect it when you pass it to Join. So instead of getting a flat list as before, you end up with nested errors. Not sure if that's what you want.

Reply:
I'm not sure if that's necessarily a problem? Even if the underlying struct is nested, the final error is printed correctly and you can use `errors.Is` to find errors within it. Example: https://go.dev/play/p/kEc0t70xrnD

We retain all previous functionality, unless we implement out own functions that search through the error, but AFAIK we don't have anything like that.